core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -287,6 +287,12 @@ default ExecutorService makeContextAware(ExecutorService executor) {
     <T extends Future<?>> GenericFutureListener<T> makeContextAware(GenericFutureListener<T> listener);
 
     /**
+     * Returns a {@link CompletableFuture} that makes sure the current {@link CompletableFuture} is set and
+     * then invokes the input {@code future}.
+     */
+    <T> CompletableFuture<T> makeContextAware(CompletableFuture<T> future);
```

Comment:
How about this:

```java
interface RequestContext {
    <T> CompletionStage<T> makeContextAware(CompletionStage<T> stage);
    default <T> CompletableFuture<T> makeContextAware(CompletableFuture<T> future) {
        return makeContextAware((CompletionStage<T>) future).toCompletableFuture();
    }
}

class AbstractRequestContext {
    final <T> CompletionStage<T> makeContextAware(CompletionStage stage) {
        ... the same code you wrote ... (it's OK because CompletableFuture implements CompletionStage.)
    }

    final <T> CompletableFuture<T> makeContextAware(CompletableFuture<T> future) {
        return RequestContext.super.makeContextAware(future);
    }
}
```